packages/core/upload/server/controllers/admin-api.js

```diff
@@ -194,7 +194,10 @@ const findEntityAndCheckPermissions = async (ability, action, model, id) => {
 
   const pm = strapi.admin.services.permission.createPermissionsManager({ ability, action, model });
 
-  const author = await strapi.admin.services.user.findOne(file[CREATED_BY_ATTRIBUTE].id, ['roles']);
+  const author = await strapi.admin.services.user.findOne(
```

Comment:
that would rather deserve to be a conditional from the looks of it. if their is no created_by you don't want to call findOne(' ') with an empty string

Reply:
I agree with you.
```suggestion
  const creatorId = _.get(file, [CREATED_BY_ATTRIBUTE, 'id'], '');
  const author = creatorId ? await strapi.admin.services.user.findOne(creatorId, ['roles']) : null;
```
What do you think?

Reply:
lgtm

Reply:
you can leave creatorId undefiend instead of empty string though by default

Reply:
Done :)